content/doc/book/pipeline/reference.adoc

```diff
@@ -0,0 +1,492 @@
+`image none` was specified.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+post {
+    always {
+        email recipient: ['one@example.com','two@example.com'], subject: "Build complete", body: "Your build has completed"
+    }
+    failure {
+        sh "bash ./cleanup-from-failure.sh"
+    }
+    success {
+        sh "git push origin master"
+    }
+}
+----
+
+[[triggers]]
+=== Triggers
+
+* _Description_: Triggers for this job, as used in other Jenkins jobs.
+* _Required_: No
+* _Allowed In_: Top-level `pipeline` closure only.
+* _Parameters_: None
+* _Takes a Closure_: Yes
+* _Closure Contents_: A sequence of one or more trigger configurations,
+using `@Symbol` names for constructors.
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `cron('@daily')` and the like.
+** Also note that the `SCMTrigger` won't work with the `scm` `@Symbol` -
+with Jenkins 2.22 or later, the `pollScm` symbol does work.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+triggers {
+    cron('@daily')
+}
+----
+
+[[build-parameters]]
+=== Build Parameters
+
+* _Description_: Build parameters that will be prompted for at build
+time.
+* _Required_: No
+* _Allowed In_: Top-level `pipeline` closure only.
+* _Parameters_: None
+* _Takes a Closure_: Yes
+* _Closure Contents_: A sequence of one or more parameter definition
+configurations, using `@Symbol` names for constructors.
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `booleanParam(...)` and the like.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+parameters {
+    booleanParam(defaultValue: true, description: '', name: 'flag')
+    string(defaultValue: '', description: '', name: 'SOME_STRING')
+}
+----
+
+[[options]]
+=== Options
+
+* _Description_: Traditional `JobProperty`, such as `buildDiscarder` or
+`disableConcurrentBuilds`, Declarative-specifc options, such as
+`skipDefaultCheckout`, and "wrappers" that should wrap the entire build,
+such as `timeout`.
+* _Required_: No
+* _Allowed In_: Top-level `pipeline` closure only.
+* _Parameters_: None
+* _Takes a Closure_: Yes
+* _Closure Contents_: A sequence of one or more Declarative option or
+job property configurations, using `@Symbol` names for constructors.
+** Note that `[$class: 'Foo', arg1: 'something', ...]` syntax can not be
+used, only `booleanParam(...)` and the like.
+** Note that the `parameters` and `pipelineTriggers` `@Symbol` cannot
+be used here directly.
+* _Examples_:
+
+[pipeline]
+----
+// Declarative //
+options {
+    buildDiscarder(logRotator(numToKeepStr:'1'))
+    disableConcurrentBuilds()
+}
+----
+
+[[declarative-subset-of-groovy]]
+=== Declarative Subset of Groovy
+
+* Top-level has to be a block
+* No semicolons as statement separators. Each statement has to be on its
+own line
+* Block must only consists of method call statements, assignment
+statements, or property reference statement
```

Comment:
parallel structure -- with two plurals in the list, the third item should also be plural, so "property reference statements"